### PR TITLE
join votes for participation check

### PR DESCRIPTION
**Pull request by dartcafe**
Count current user's votes via join in poll
regarding #3362

### CODE REVIEW

File: lib/AppInfo/Application.php [hunk 1]
```diff
@@ -132,4 +149,66 @@ public function register(IRegistrationContext $context): void {
...
+	private function registerServices(IRegistrationContext $context) {
+		$context->registerService(UserMapper::class, function (ContainerInterface $c): UserMapper {
+			return new UserMapper(
+				$c->get(IDBConnection::class),
+				$c->get(ISession::class),
+				$c->get(IUserSession::class),
+				$c->get(IUserManager::class),
+				$c->get(LoggerInterface::class),
+			);
+		});
```
Comment (come-nc) [inline lib/AppInfo/Application.php:165]: Is all this needed? This is the default DI behavior to my knowledge.

Comment (dartcafe) [reply lib/AppInfo/Application.php:165]: I was annoyed by the unit tests, where every deopendency needed to be instatiated. 
If there is a DI possibility for the tests, this is indeed unnecessary.

Comment (come-nc) [reply lib/AppInfo/Application.php:165]: Usually you use Mocks for the tests

Comment (dartcafe) [reply lib/AppInfo/Application.php:165]: Yeah. Tests. See https://github.com/nextcloud/polls/issues/3147 😉

**Pull request merged**